src/test/java/com/timgroup/statsd/UnixDatagramSocketTest.java

```diff
@@ -0,0 +1,188 @@
+            @Override
+            public SocketAddress call() throws Exception {
+                return new UnixSocketAddressWithTransport(new UnixSocketAddress(socketFile.getPath()), UnixSocketAddressWithTransport.TransportType.UDS_DATAGRAM);
+            }
+        };
+
+        client = new NonBlockingStatsDClientBuilder().prefix("my.prefix")
+                .addressLookup(addressLookup)
+                .port(0)
+                .queueSize(1)
+                .timeout(1)  // non-zero timeout to ensure exception triggered if socket buffer full.
+                .connectionTimeout(100)
+                .socketBufferSize(1024 * 1024)
+                .enableAggregation(false)
+                .errorHandler(this)
+                .originDetectionEnabled(false)
+                .build();
+
+        clientAggregate = new NonBlockingStatsDClientBuilder().prefix("my.prefix")
+                .addressLookup(addressLookup)
+                .port(0)
+                .queueSize(1)
+                .timeout(1)  // non-zero timeout to ensure exception triggered if socket buffer full.
+                .connectionTimeout(100)
+                .socketBufferSize(1024 * 1024)
+                .enableAggregation(false)
+                .errorHandler(this)
+                .originDetectionEnabled(false)
+                .build();
+    }
+
+    @After
+    public void stop() throws Exception {
+        client.stop();
+        clientAggregate.stop();
+        server.close();
+    }
+
+    @Test
+    public void assert_default_uds_size() throws Exception {
+        assertEquals(client.statsDProcessor.bufferPool.getBufferSize(), NonBlockingStatsDClient.DEFAULT_UDS_MAX_PACKET_SIZE_BYTES);
+    }
+
+    @Test(timeout = 5000L)
+    public void sends_to_statsd() throws Exception {
+        for(long i = 0; i < 5 ; i++) {
+            client.gauge("mycount", i);
+            server.waitForMessage();
+            String expected = String.format("my.prefix.mycount:%d|g", i);
+            assertThat(server.messagesReceived(), contains(expected));
+            server.clear();
+        }
+        assertThat(lastException.getMessage(), nullValue());
+    }
+
+    @Test(timeout = 10000L)
+    public void resist_dsd_restart() throws Exception {
+        // Send one metric, check that it works.
+        client.gauge("mycount", 10);
+        server.waitForMessage();
+        assertThat(server.messagesReceived(), contains("my.prefix.mycount:10|g"));
+        server.clear();
+        assertThat(lastException.getMessage(), nullValue());
+
+        // Close the server, client should throw an IOException
+        server.close();
+        while(lastException.getMessage() == null) {
+            client.gauge("mycount", 20);
+            Thread.sleep(10);
+        }
+        // Depending on the state of the client at that point we might get different messages.
+        assertThat(lastException.getMessage(), anyOf(containsString("Connection refused"), containsString("Broken pipe")));
```

Comment:
I think this wasn't in the datagram tests previously.

It is quite difficult to review this way. Could you also please make it so that these old tests don't appear as all newly added? Keeping the old filename for the test would be sufficient, I think.

Reply:
Fixing